metricbeat/module/system/network/network.go

```diff
@@ -91,11 +95,44 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 		isOpen := r.Event(mb.Event{
 			MetricSetFields: ioCountersToMapStr(counters),
 		})
+
+		// accumulate values from all interfaces
+		networkInBytes += counters.BytesRecv
+		networkOutBytes += counters.BytesSent
+		networkInPackets += counters.PacketsRecv
+		networkOutPackets += counters.PacketsSent
+
 		if !isOpen {
-			return nil
+			continue
 		}
 	}
 
+	if len(m.prevCounters) != 0 {
+		// convert network metrics from counters to gauges
+		r.Event(mb.Event{
+			RootFields: common.MapStr{
+				"host": common.MapStr{
+					"network": common.MapStr{
+						"in": common.MapStr{
+							"bytes":   networkInBytes - m.prevCounters["networkInBytes"],
+							"packets": networkInPackets - m.prevCounters["networkInPackets"],
+						},
+						"out": common.MapStr{
+							"bytes":   networkOutBytes - m.prevCounters["networkOutBytes"],
+							"packets": networkOutPackets - m.prevCounters["networkOutPackets"],
+						},
+					},
+				},
+			},
+		})
+	}
+
+	// update prevCounters
+	m.prevCounters["networkInBytes"] = networkInBytes
+	m.prevCounters["networkInPackets"] = networkInPackets
+	m.prevCounters["networkOutBytes"] = networkOutBytes
+	m.prevCounters["networkOutPackets"] = networkOutPackets
```

Comment:
Nit. We could maybe use structs to store the previous counters, what would be slightly more efficient and less error prone.